File: src/leetcode/ShuffleAnArray.java
package leetcode;

import java.util.Random;

public class ShuffleAnArray {
	public int nums[];
	public Random random;
	
	public void Solution(int[] nums) {
        this.nums = nums;
    }
    
    /** Resets the array to its original configuration and return it. */
    public int[] reset() {
        return nums;
    }
    
    /** Returns a random shuffling of the array. */
    public int[] shuffle() {
        int[] random = new int[nums.length];
        for(int i = 0; i < nums.length; i++) {
        	int r = (int) (Math.random() * (i + 1));
        	random[i] = random[r];
        	random[r] = nums[i];
        }
        return random;
    }
}

File: src/leetcode/ShuffleAnArray.py
import random
class Solution(object):

    def __init__(self, nums):
        """
        :type nums: List[int]
        """
        self.nums = nums

    def reset(self):
        """
        Resets the array to its original configuration and return it.
        :rtype: List[int]
        """
        return self.nums

    def shuffle(self):
        """
        Returns a random shuffling of the array.
        :rtype: List[int]
        """
        shuff = self.nums[:]
        for i in range(len(shuff)-1, 0, -1):
            j = random.randint(0, i+1)
            shuff[i], shuff[j] = shuff[j], shuff[i]
            return shuff
if __name__ == "__main__":
    print 1
